manifests/auth.pp

```diff
@@ -23,61 +23,70 @@
 # @param cracklib_retry
 # @param deny
 # @param display_account_lock
-# @param enable_separator
 # @param fail_interval
 # @param remember
 # @param remember_retry
 # @param remember_for_root
+# @param even_deny_root
 # @param root_unlock_time
+# @param hash_algorithm
 # @param rounds
 # @param uid
 # @param unlock_time
 # @param preserve_ac
 # @param use_netgroups
 # @param use_openshift
-# @param separator
 # @param sssd
 # @param tty_audit_users
+# @param separator
+# @param enable_separator
 # @param content
 #
 define pam::auth (
   Pam::PasswordBackends  $password_check_backend    = $::pam::password_check_backend,
   Boolean                $cracklib_enforce_for_root = $::pam::cracklib_enforce_for_root,
   Boolean                $cracklib_reject_username  = $::pam::cracklib_reject_username,
-  Integer                $cracklib_retry            = $::pam::cracklib_retry,
-  Optional[Integer]      $cracklib_difok            = $::pam::cracklib_difok,
-  Optional[Integer]      $cracklib_maxrepeat        = $::pam::cracklib_maxrepeat,
-  Optional[Integer]      $cracklib_maxsequence      = $::pam::cracklib_maxsequence,
-  Optional[Integer]      $cracklib_maxclassrepeat   = $::pam::cracklib_maxclassrepeat,
+  Optional[Integer[0]]   $cracklib_difok            = $::pam::cracklib_difok,
+  Optional[Integer[0]]   $cracklib_maxrepeat        = $::pam::cracklib_maxrepeat,
+  Optional[Integer[0]]   $cracklib_maxsequence      = $::pam::cracklib_maxsequence,
+  Optional[Integer[0]]   $cracklib_maxclassrepeat   = $::pam::cracklib_maxclassrepeat,
   Optional[Boolean]      $cracklib_gecoscheck       = $::pam::cracklib_gecoscheck,
-  Optional[Integer]      $cracklib_dcredit          = $::pam::cracklib_dcredit,
-  Optional[Integer]      $cracklib_ucredit          = $::pam::cracklib_ucredit,
-  Optional[Integer]      $cracklib_lcredit          = $::pam::cracklib_lcredit,
-  Optional[Integer]      $cracklib_ocredit          = $::pam::cracklib_ocredit,
-  Optional[Integer]      $cracklib_minclass         = $::pam::cracklib_minclass,
-  Optional[Integer]      $cracklib_minlen           = $::pam::cracklib_minlen,
-  Integer                $deny                      = $::pam::deny,
+  Optional[Integer[-1]]  $cracklib_dcredit          = $::pam::cracklib_dcredit,
```

Comment:
I think the lower bound of dcredit, ucredit, lcredit, and ocredit should be much smaller (e.g., -5) to give the user the ability to configure more restrictive enforcement.